src/main/scala/firrtl/Compiler.scala

```diff
@@ -540,7 +540,10 @@ trait Compiler extends Transform with DependencyAPIMigration {
     * @param customTransforms Any custom [[Transform]]s that will be inserted
     *   into the compilation process by [[CompilerUtils.mergeTransforms]]
     */
-  @deprecated("Please use compileAndEmit or other compile method instead", "firrtl 1.0")
+  @deprecated(
+    "Migrate to '(new FirrtlStage).execute(args: Array[String], annotations: AnnotationSeq)'." +
+      "This will be removed in 1.4.",
```

Comment:
Not `1.5` since `1.3` is already released? Shouldn't the deprecation go through one full release cycle?

Reply:
I was hoping that this would be fine. Since all `Compiler`s are deprecated in 1.3 and to-be-removed in 1.4 (though this may not happen...), I figured it would be fine to retcon deprecations on their methods just to make it more clear about the migration path.

What do you think?

More info: removing `Compiler` is a serious pain and I'm not sure I have the bandwidth before 1.4 goes out. This requires removing `CompilerAnnotation` (which isn't deprecated) to using only `RunFirrtlTransformAnnotation`s. E.g., `-X <compiler>` should map to dependencies and not a compiler.

I'm fine with 1.5. Though, that should probably bump the overall Compiler removal timeline, too.

Reply:
Oh, ok, so this PR is mostly intended as a backport for `1.3` to clarify your intention?
Then that's fine!

Reply:
Yeah, I was planning to get this into 1.3.3 (I updated the deprecation messages to more accurately indicate "since FIRRTL 1.3.3" for the other two deprecations that are new).